catalog/ui/catalog-ui-search/src/main/webapp/component/sort-item/sort-item.view.js

```diff
@@ -21,11 +21,55 @@ define([
     'js/CustomElements',
     'component/singletons/metacard-definitions',
     'component/dropdown/dropdown.view',
+    'component/property/property.view',
+    'component/property/property',
     'properties'
-], function(Marionette, _, $, template, CustomElements, metacardDefinitions, DropdownView, properties) {
+], function (Marionette, _, $, template, CustomElements, metacardDefinitions, DropdownView, PropertyView, Property, properties) {
 
     var blacklist = ['anyText', 'anyGeo'];
 
+    function getSortLabel(attribute) {
+        var ascendingLabel, descendingLabel;
+        if (metacardDefinitions.metacardTypes[attribute] === undefined) {
+            ascendingLabel = descendingLabel = '';
+        } else {
+            switch (metacardDefinitions.metacardTypes[attribute].type) {
+                case 'DATE':
+                    ascendingLabel = 'Earliest';
+                    descendingLabel = 'Latest';
+                    break;
+                case 'BOOLEAN':
+                    ascendingLabel = 'True First'; //Truthiest
+                    descendingLabel = 'False First'; //Falsiest
+                    break;
+                case 'LONG':
+                case 'DOUBLE':
+                case 'FLOAT':
+                case 'INTEGER':
+                case 'SHORT':
+                    ascendingLabel = 'Smallest';
+                    descendingLabel = 'Largest';
+                    break;
+                case 'STRING':
+                    ascendingLabel = 'A to Z';
+                    descendingLabel = 'Z to A';
+                    break;
+                case 'GEOMETRY':
+                case 'XML':
+                case 'BINARY':
+                default:
```

Comment:
I think there were plans to make it smarter --  but the basis of it is it will at least group all fields with identical values together.